File: Medium/2-100/Prob_62/uniquepath.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        row = [1]*n
        for _ in range(m-2,-1,-1):
            for c in range(n-2,-1,-1):
                row[c] = row[c]+row[c+1]
        return row[0]

File: Medium/2-100/Prob_62/UniquePath.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function (m, n) {
  let row = new Array(n).fill(1);
  for (let r = m - 2; r > -1; r--) {
    for (let c = n - 2; c > -1; c--) {
      row[c] += row[c + 1];
    }
  }
  return row[0];
};
